src/megameklab/com/util/MenuBarCreator.java

```diff
@@ -607,198 +591,119 @@ private void jMenuGetUnitWeightBreakdownFromCache_actionPerformed() {
-        } finally {
+            Entity tempEntity = new MechFileParser(unitFile).getEntity();
             tempEntity.calculateBattleValue(true, true);
             UnitUtil.showBVCalculations(tempEntity.getBVText(), parentFrame);
+        } catch (Exception ex) {
+            JOptionPane.showMessageDialog(parentFrame,
+                    String.format(resourceMap.getString("message.invalidUnit.format"),
+                            ex.getMessage()));
         }
 
     }
 
     private void jMenuGetUnitValidationFromFile_actionPerformed() {
-
-        Entity tempEntity = null;
-        String filePathName = System.getProperty("user.dir").toString() + "/data/mechfiles/";
-        File unitFile = new File(filePathName);
-        JFileChooser f = new JFileChooser(filePathName);
-        f.setLocation(parentFrame.getLocation().x + 150, parentFrame.getLocation().y + 100);
-        f.setDialogTitle("Choose Unit");
-        f.setDialogType(JFileChooser.OPEN_DIALOG);
-        f.setMultiSelectionEnabled(false);
-
-        FileNameExtensionFilter filter = new FileNameExtensionFilter("Unit Files", "blk", "mtf", "hmp");
-
-        // Add a filter for mul files
-        f.setFileFilter(filter);
-
-        int returnVal = f.showOpenDialog(parentFrame);
-        if ((returnVal != JFileChooser.APPROVE_OPTION) || (f.getSelectedFile() == null)) {
-            // I want a file, y'know!
-            return;
-        }
-        unitFile = f.getSelectedFile();
+        File unitFile = loadUnitFile();
+        if (unitFile == null) return;
 
         try {
-            tempEntity = new MechFileParser(unitFile).getEntity();
-        } catch (Exception ex) {
-            JOptionPane.showMessageDialog(parentFrame, String.format("Warning:Invalid unit, it might load incorrectly!\n%1$s", ex.getMessage()));
-        } finally {
+            Entity tempEntity = new MechFileParser(unitFile).getEntity();
             UnitUtil.showValidation(tempEntity, parentFrame);
+        } catch (Exception ex) {
+            JOptionPane.showMessageDialog(parentFrame,
+                    String.format(resourceMap.getString("message.invalidUnit.format"),
+                    ex.getMessage()));
         }
     }
 
-    private void jMenuGetUnitBreakdownFromFile_actionPerformed() {
-
-        Entity tempEntity = null;
-        String filePathName = System.getProperty("user.dir").toString() + "/data/mechfiles/";
-        File unitFile = new File(filePathName);
-        JFileChooser f = new JFileChooser(filePathName);
-        f.setLocation(parentFrame.getLocation().x + 150, parentFrame.getLocation().y + 100);
-        f.setDialogTitle("Choose Unit");
-        f.setDialogType(JFileChooser.OPEN_DIALOG);
-        f.setMultiSelectionEnabled(false);
-
-        FileNameExtensionFilter filter = new FileNameExtensionFilter("Unit Files", "blk", "mtf", "hmp");
+    private @Nullable
+    File loadUnitFile() {
+        String filePathName = System.getProperty("user.dir") + "/data/mechfiles/";
```

Comment:
Does Java have a path combiner? I know in .NET we'd do: `Path.Combine(System.getProperty("user.dir"), "data", "mechfiles")` and never care about path separators.

Reply:
The java.nio.file.* package has tools that do that sort of thing, but I don't use it often enough to know off the top of my head. This line was moved from elsewhere in the code when I extracted a method to consolidate identical code that appeared 3-4 places in the class.